Controllers/FileController.cs

```diff
@@ -1,20 +1,77 @@
 using Microsoft.AspNetCore.Mvc;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Collections.Generic; 
+using System.Globalization;
+
 namespace TodoApi.Controllers
 {
+    public class Author
+    {
+        public string Name { get; set; }
+        public string CountryOfOrigin { get; set; }
+        public DateTime LastSubmissionTimestamp { get; set; }
+    }
+    public class FormData
+    {
+        public string[]? Files { get; set; }
+        public Dictionary<string, int>? DessertVotes { get; set; }
+        public string? CurrentWeather { get; set; }
+        public List<Author>? AuthorsList { get; set; }
+    }
+
     [Route("api/[controller]")]
     [ApiController]
     public class FileController : ControllerBase
     {
+        
+        private const string FilePath = "formDataStorage.json";
+        private Dictionary<string, int>? dessertVotes = new Dictionary<string, int>();
+        private string? currentWeather = "clear";
+        private List<Author> authorsList = new List<Author> {};
+        private dynamic allData;
+
+        private async Task InitializeFormTypes()
+        {
+            if (System.IO.File.Exists(FilePath))
+            {
+                var json = await System.IO.File.ReadAllTextAsync(FilePath);
+                Console.WriteLine(json);
+                allData = JsonSerializer.Deserialize<FormData>(json);
+                dessertVotes = allData?.DessertVotes ?? new Dictionary<string, int>();
+                authorsList = allData?.AuthorsList ?? new List<Author>();
+                currentWeather = allData?.CurrentWeather ?? "clear";
+            }
+            else
+            {
+                dessertVotes = new Dictionary<string, int>()
+                {
+                    {"Red Velvet Cake", 0},
+                    {"Peach Pie", 0},
+                    {"Lemon Bars", 0}
+                };
+                authorsList = new List<Author>();
+                currentWeather = "clear";
+            }
+        }
+
+        public FileController()
+        {
+            InitializeFormTypes().Wait();
```

Comment:
Probably better to wait with a timeout so if it hangs, we at least get a 500 error on the client side